server/routes/routes.js

```diff
@@ -0,0 +1,79 @@
+const express = require('express');
+const userService = require('../../server/services/user');
+const roleService = require('../../server/services/role');
+const documentService = require('../../server/services/doc');
+const validate = require('../middlewares/auth');
+
+const router = express.Router();
+
+// Setup default route
+router.get('/', (req, res) => {
+  res.status(200).send({
+    message: 'Welcome to the Document Management App!'
+  });
+});
+
+// Route to create and retreive users.
+router.route('/users')
+  .post(userService.createUser)
+  .get(validate.validateAdmin, userService.getAllUsers);
+
+// Route to retrieve single user data
+router.route('/users/:username')
+  .get(validate.validateToken, userService.getUser)
+  .put(validate.validateToken, userService.updateUser)
+  .delete(validate.validateAdmin, userService.deleteUser);
+
+// User login route
+router.route('/login')
+  .get(userService.login);
```

Comment:
Is this not supposed to be a `POST` because this implies that username and password will have to be part of the url